cartographer_rviz/cartographer_rviz/submaps_display.cc

```diff
@@ -162,20 +165,31 @@ void SubmapsDisplay::processMessage(
 
 void SubmapsDisplay::update(const float wall_dt, const float ros_dt) {
   ::cartographer::common::MutexLocker locker(&mutex_);
-  // Update the fading by z distance.
-  try {
-    const ::geometry_msgs::TransformStamped transform_stamped =
-        tf_buffer_.lookupTransform(map_frame_property_->getStdString(),
-                                   tracking_frame_property_->getStdString(),
-                                   ros::Time(0) /* latest */);
-    for (auto& trajectory : trajectories_) {
-      for (auto& submap_entry : trajectory.second) {
-        submap_entry.second->SetAlpha(
-            transform_stamped.transform.translation.z);
+  if (map_frame_ != nullptr) {
+    // Update the map frame to fixed frame transform.
+    Ogre::Vector3 position;
+    Ogre::Quaternion orientation;
+    if (context_->getFrameManager()->getTransform(
+            *map_frame_, ros::Time(0) /* latest */, position, orientation)) {
+      map_node_->setPosition(position);
+      map_node_->setOrientation(orientation);
+      context_->queueRender();
+    }
+    // Update the fading by z distance.
+    try {
+      const ::geometry_msgs::TransformStamped transform_stamped =
+          tf_buffer_.lookupTransform(*map_frame_,
+                                     tracking_frame_property_->getStdString(),
+                                     ros::Time(0) /* latest */);
```

Comment:
would it be better to use a local var with ros::Time(0) to use the same timestamp here and above.